include/classes/DAO/DAO.class.php

```diff
@@ -63,22 +63,29 @@ function DAO()
 	function execute($sql)
 	{
 		$sql = trim($sql);
-		$result = mysql_query($sql, $this->db) or die($sql . "<br />". mysql_error());
+		$result = mysqli_query($this->db, $sql) or die($sql . "<br />". mysqli_error($this->db));
 
 		// Deal with "select" statement: return false if no row is returned, otherwise, return an array
 		if ($result !== true && $result !== false) {
 			$rows = false;
 			
-			while ($row = mysql_fetch_assoc($result)){
+			while ($row = mysqli_fetch_assoc($result)){
 				if (!$rows) $rows = array();
 				
 			    $rows[] = $row;
 			}
-			mysql_free_result($result);
+			mysqli_free_result($result);
 			return $rows;
 		}
 		return true;
 	}
 
+	function addSlashes($sql){
+		return mysqli_real_escape_string($this->db,$sql);
```

Comment:
This function should also check `magic_quotes_gpc`: when this directive is on, the received data already have slashes added and those slashes should be firstly stripped before calling `mysqli_real_escape_string()`. So the logic of this function should be as the [old logic](https://github.com/inclusive-design/AChecker/blob/master/include/vitals.inc.php#L100-L106) that creates `$addslashes` in vitals.inc.php.

Reply:
Ok, would do that